### PR TITLE
Update development & documentation requirements to latest version

### PR DESCRIPTION
**Pull request by gsheni**
- You can preview the docs here:
- https://readthedocs.com/projects/feature-labs-inc-featuretools/builds/1044856/

### CODE REVIEW

**Comment by gsheni** [comment]
I was originally going to use a .readthedocs.yml file as below. However, this resulted in a significant increase in ru ntime 455 seconds to 594 seconds. Therefore I kept the existing format. It seems strange that readthedocs has the Spark requirements, graphviz, and pandoc installed (without specifying it in .readthedocs.yaml)

```yaml
# .readthedocs.yaml
# Read the Docs configuration file
# See https://docs.readthedocs.io/en/stable/config-file/v2.html for details

# Required
version: 2

# Set the version of Python and other tools you might need
build:
  os: ubuntu-20.04
  tools:
    python: "3.8"
  apt_packages:
    - pandoc
    - graphviz
    - openjdk-11-jre-headless
  jobs:
    post_create_environment:
      - export PYARROW_IGNORE_TIMEZONE=1
      - export JAVA_HOME="/usr/lib/jvm/java-11-openjdk-amd64"

# Build documentation in the docs/ directory with Sphinx
sphinx:
  configuration: docs/source/conf.py

# Optionally build your docs in additional formats such as PDF and ePub
formats: []

# Optionally set the version of Python and requirements required to build your docs
python:
  install:
    - method: pip
      path: .
      extra_requirements:
        - docs
```

**Comment by gsheni** [comment]
- The codecov error can be ignored

File: release.md [hunk 1]
```diff
@@ -82,18 +82,21 @@ After the release pull request has been merged into the `main` branch, it is tim
...
-    * You can also create the PR by clicking the branch name (ex - `conda-autocreate-v0.13.3`): 
+    * You will need to visit this URL, and create a pull request.
+    * You can also create the pull request by clicking the branch name (e.g. - `v0.13.3`): 
       - https://github.com/alteryx/featuretools-feedstock/branches
-5. Verify the `requirements['run']` (in __recipe/meta.yml__) match the `install_requires` in __featuretools/setup.cfg__.
-    * Verify the `test['requires']` (in __recipe/meta.yml__) match the test requirements in `[options.extras_require]` in __featuretools/setup.cfg__
+5. Verify that the PR has the following: 
+    * The `build['number']` is 0 (in __recipe/meta.yml__).
+    * The `requirements['run']` (in __recipe/meta.yml__) match the `install_requires` in __woodwork/setup.cfg__.
+    * The `test['requires']` (in __recipe/meta.yml__) match the test requirements in `[options.extras_require]` in __woodwork/setup.cfg__
```
Comment (tamargrey) [inline release.md:98]: __woodwork --> __featuretools here and the line above (

Comment (gsheni) [reply release.md]: Fixed
https://github.com/alteryx/featuretools/pull/2225/commits/fc66dad2f6e08deb2adfbcc28d7e463b7aec17a1

File: release.md [hunk 2]
```diff
@@ -82,18 +82,21 @@ After the release pull request has been merged into the `main` branch, it is tim
...
 
-In order to release on conda-forge, you can either wait for a bot to create a PR, or manually kickoff the creation with GitHub Actions
+In order to release on conda-forge, you can either wait for a bot to create a pull request, or manually kickoff the creation with GitHub Actions
 
-### Option 1: Manually create the new PR with GitHub Actions
+### Option 1: Manually create the new pull request with GitHub Actions
 1. Go to this GitHub Action: https://github.com/alteryx/featuretools/actions/workflows/create_feedstock_pr.yaml
-2. Input the released version with the v prefix (e.g. v0.13.3)
-3. Kickoff the GitHub action, and monitor the Job Summary.
+2. Click **Run workflow** and input the released version with the v prefix (e.g. `v0.13.3`)
```
Comment (sbadithe) [inline release.md:89]: Very minor but I find the following a little more clear: 
```suggestion
2. Click **Run workflow** and input the letter `v` followed by the release version (e.g. `v0.13.3`)
```

File: .github/workflows/build_docs.yaml [hunk 3]
```diff
@@ -5,6 +5,9 @@ on:
     branches:
       - main
 
+env:
+  PYARROW_IGNORE_TIMEZONE: 1
```
Comment (gsheni) [inline .github/workflows/build_docs.yaml:9]:
```suggestion
  PYARROW_IGNORE_TIMEZONE: 1
  JAVA_HOME: "/usr/lib/jvm/java-11-openjdk-amd64"
```

File: release.md [hunk 4]
```diff
@@ -82,18 +82,21 @@ After the release pull request has been merged into the `main` branch, it is tim
...
+3. Kickoff the GitHub Action, and monitor the Job Summary.
 4. At the completion of the job, you should see summary output, with a URL. 
-    * You will need to visit this URL, and create a PR.
-    * You can also create the PR by clicking the branch name (ex - `conda-autocreate-v0.13.3`): 
+    * You will need to visit this URL, and create a pull request.
```
Comment (sbadithe) [inline release.md:92]: Just to mirror the structure of the other steps
```suggestion
    * Visit this URL and create a pull request.
```

File: release.md [hunk 5]
```diff
@@ -82,18 +82,21 @@ After the release pull request has been merged into the `main` branch, it is tim
...
+    * You can also create the pull request by clicking the branch name (e.g. - `v0.13.3`):
```
Comment (sbadithe) [inline release.md:93]:
```suggestion
    * Alternatively, create the pull request by clicking the branch name (e.g. - `v0.13.3`): 
```

File: docs/source/release_notes.rst [hunk 6]
```diff
@@ -13,6 +13,7 @@ Future Release
     * Documentation Changes
     * Testing Changes
         * Update create feedstock PR forked repo to use (:pr:`2223`)
+        * Update development requirements and use latest for documentation (:pr:`2225`)
 
     Thanks to the following people for contributing to this release:
     :user:`gsheni`, :user:`ozzieD`, user:`sbadithe`
```
Comment (rwedge) [inline docs/source/release_notes.rst:19]:
```suggestion
    :user:`gsheni`, :user:`ozzieD`, :user:`sbadithe`
```

**Review by rwedge — APPROVED** [review]
LGTM, left one suggested change

**Pull request merged**